testsuite/functional-testsuite/src/test/ft-resources/testsuite-oss/system/system_capabilities.t.js

```diff
@@ -0,0 +1,120 @@
+/*
+ * Sonatype Nexus (TM) Open Source Version
+ * Copyright (c) 2008-present Sonatype, Inc.
+ * All rights reserved. Includes the third-party code listed at http://links.sonatype.com/products/nexus/oss/attributions.
+ *
+ * This program and the accompanying materials are made available under the terms of the Eclipse Public License Version 1.0,
+ * which accompanies this distribution and is available at http://www.eclipse.org/legal/epl-v10.html.
+ *
+ * Sonatype Nexus (TM) Professional Version is available from Sonatype, Inc. "Sonatype" and "Sonatype Nexus" are trademarks
+ * of Sonatype, Inc. Apache Maven is a trademark of the Apache Software Foundation. M2eclipse is a trademark of the
+ * Eclipse Foundation. All other trademarks are the property of their respective owners.
+ */
+/**
+ * Tests system capabilities CRUD.
+ */
+StartTest(function(t) {
+  var waitForStoreToLoad = {
+    waitFor: 'storesToLoad', args: function() {
+        return t.cq1('nx-coreui-capability-list').getStore();
+    }
+  },
+      // Name of Capability tested
+      name = 'Base URL: Force';
+
+  t.describe('Given an admin when the admin is administering Capabilities', function(t) {
+    t.it('then CRUD should work', function(t) {
```

Comment:
I realized this is dated and fixed and closed but I remember setting this
up because I thought each .it had to be able to run independantly and the
test Jeffry and I wrote update is not created by default so cannot be run
without create.
I checked Sencha and they say "All specs of the same nesting level are
executed sequentially." (ref:
http://www.bryntum.com/docs/siesta/#!/api/Siesta.Test.BDD-method-it)
Does that not say the same thing?

I am not sure where else I would have gleaned that (lots of info=)) but if
that's not true (or is just a style thing), then I agree it is easier to
follow and can easily do this style in the future.

Thanks for cleaning up after me(us), Kelly.
Peace,
Joe

On Mon, Jul 6, 2015 at 12:54 PM, Stuart McCulloch <notifications@github.com>
wrote:

> In
> testsuite/functional-testsuite/src/test/ft-resources/testsuite-oss/system/system_capabilities.t.js
> <https://github.com/sonatype/nexus-oss/pull/1522#discussion_r33955784>:
>
> > + * Eclipse Foundation. All other trademarks are the property of their respective owners.
> > + */
> > +/**
> > + * Tests system capabilities CRUD.
> > + */
> > +StartTest(function(t) {
> > +  var waitForStoreToLoad = {
> > +    waitFor: 'storesToLoad', args: function() {
> > +        return t.cq1('nx-coreui-capability-list').getStore();
> > +    }
> > +  },
> > +      // Name of Capability tested
> > +      name = 'Base URL: Force';
> > +
> > +  t.describe('Given an admin when the admin is administering Capabilities', function(t) {
> > +    t.it('then CRUD should work', function(t) {
>
> Other CRUD-based tests separate out each operation into different it
> methods (for example see
> https://github.com/sonatype/nexus-oss/blob/master/testsuite/functional-testsuite/src/test/ft-resources/testsuite-oss/repository/repository_crud.t.js#L51)
> which is imho easier to follow.
>
> —
> Reply to this email directly or view it on GitHub
> <https://github.com/sonatype/nexus-oss/pull/1522/files#r33955784>.
>